### ISSUE TITLE
How to listen file descriptor, perhaps using systemd activation protocol

### ISSUE THREAD

**Issue by tailhook** [envoyproxy/envoy#3596]
It would be nice to be able to listen standard port 80 and 443 without starting envoy as root user. Systemd provides the standard protocol for that called [socket activation protocol](http://0pointer.de/blog/projects/socket-activation.html). There are also other ways to do the same with e.g. nginx even if it doesn't support that particular protocol.

Is there any equivalent of passing fd to envoy?

If you're not familiar with the concept. It's something like this: supervisor (for example systemd) binds a socket at port say 80. Then it puts the file descriptor into fixed number usually 3 (or starting from three if there are multiple) and removes CLOEXEC flag. So when process starts it has normal stdio file descriptors 0 (stdin), 1 (stdout), 2 (stderr), and also ``3`` which is a listening socket and can use the latter as a normal socket.

**Comment by mattklein123** [comment]
This is not supported today. I've marked "design proposal" wanted since I think it would be good to discuss whether we want to implement something like this or not.

**Comment by tailhook** [comment]
Wow! Bots are quite bold here :)

Using it as a chance to ask how it's going on? I think the feature is quite simple and should not add any substantial maintenance burden.

Let me know if I can help somehow.

**Comment by htuch** [comment]
@tailhook yeah, this actually seems quite reasonable to me. We (Google) have some use cases that would benefit from having prebound listener FDs passed. In our case, we want these to be delivered over UDS, but I think there could be common ground.

**Comment by eriksjolund** [comment]
A side-note: Container runtime support for _socket activation_ has improved lately.
[Podman](https://github.com/containers/podman/) version 3.4.0 (released September 2021)
supports running OCI containers (i.e. Docker containers) with _socket activation_.

I wrote a [minimal demo](https://github.com/eriksjolund/mariadb-podman-socket-activation/) for a MariaDB container. (That demo is not related to Envoyproxy though).

**Comment by eriksjolund** [comment]
Interestingly, the communication over a activated socket will run with [native network performance](https://github.com/containers/podman/blob/main/docs/tutorials/socket_activation.md#native-network-performance-over-the-socket-activated-socket) when running rootless Podman. Network traffic is normally passed through the program __slirp4netns__ which comes with a performance penalty.
